### PR TITLE
fix: output now displays dependencies if they have immediate or transitive dependencies

### PR DESCRIPTION
**Pull request by jsamar17**

### CODE REVIEW

File: src/output-util.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  packageTree.dependencies.forEach((dep) => {
+    totalDetections +=
+        getNumberOfTransitiveDetectionsRec(dep, countedDependencies);
+  });
+  return totalDetections;
+
+
+  function getNumberOfTransitiveDetectionsRec(
+      packageTree: PackageTree<PointOfInterest[]>,
+      countedDependencies: Map<string, PackageTree<PointOfInterest[]>>):
```
Comment (jplaisted) [inline src/output-util.ts]: See my comment from your other PR. No need for this inner function.

File: src/output.ts [hunk 2]
```diff
@@ -44,18 +48,47 @@ function flattenPackageTree(
...
+      packageTree.data.forEach((dataPoint) => {
+        arrOfStrings.push(`     ${dataPoint.type} found in ${
+            dataPoint.fileName.split('node_modules/')[1]} at ${
+            JSON.stringify(dataPoint.position, null, 1)}`);
+      });
+    } else {
+      if (packageTree.data.length > 0) {
+        arrOfStrings.push(`  Detected Patterns:`);
+        const squashedDetections = squashDetections(packageTree);
+        for (const type of squashedDetections) {
```
Comment (jplaisted) [inline src/output.ts]: To make this a little clearer, maybe destructure?

`for (const [key, value] of squashedDetections) {`

Comment (jplaisted) [reply src/output.ts]: With maybe better names than `key` and `value`

File: src/package-tree.ts [hunk 3]
```diff
@@ -190,13 +190,13 @@ export async function resolvePaths(
...
+
+  // Checks to see if parentPath/node_modules/moduleName exists. If not call
+  // findPath on the directory above
   if ((await util.exists(moduleFolder)) &&
       (await util.stat(moduleFolder)).isDirectory()) {
-    const filesInFolder = await util.readdir(moduleFolder);
-    if (filesInFolder.includes(mod)) {
-      return path.join(moduleFolder, mod);
-    }
+    return moduleFolder;
```
Comment (jplaisted) [inline src/package-tree.ts]: Ditto, this seems to be not really a part of this PR?

**Review by kjin — APPROVED** [review]

**Comment by kjin** [comment]
Backup from before squash: https://github.com/google/node-dependency-analysis/tree/backup/jsamar17/outputUpdate

**Pull request merged**